File: laicode/q82/stack.cpp
#include <iostream>

using namespace std;

class Solution {

public:
    void removeAdjacentIV(char* s) {
        int top = -1;
        for (int i = 0; s[i]; i++) {
            if (top == -1 || s[i] != s[top]) {
                s[++top] = s[i];
            } else {
                while (top != -1 && s[top] == s[i]) {
                    i++;
                }
                i--;
                top--;
            }
        }
        s[top + 1] = '\0';
    }
};

int main() {
    Solution sol;
    char s[] = "abbbbaaccz";
    sol.removeAdjacentIV(s);
    cout << s;
    return 0;
}

File: laicode/q82/Stack.java
package laicode.q82;

public class Stack {
    public int removeAdjacentIV(char[] array) {
        
        int top = -1;
        for (int i = 0; i < array.length; i++) {
            if (top == -1 || array[top] != array[i]) {
                array[++top] = array[i];
            } else {
                while (top != -1 && array[i] == array[top]) {
                    i++;
                }
                top--;
                i--;
            }
        }
        return top + 1;
    }

    public static void main(String[] args) {
        Stack sol = new Stack();
        char[] array = "abbbaaccz".toCharArray();
        System.out.println(sol.removeAdjacentIV(array));
    }
}
